lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2326,5 +2397,25 @@ export class UserAgentApplication {
 
   }
 
+  /**
+   * Construct 'tokenRequest' from the available data in adalIdToken
+   * @param extraQueryParameters
+   */
+  private getTokenRequest(extraQueryParameters: string): AuthenticationParameters {
```

Comment:
something that denotes use this if adalTokenExists

Reply:
Cannot specifically tie this up with idToken as this is a util that builds request from exisiting client-id as scope, default authority, user and extraQueryParams. I am changing the name to buildIDTokenRequest(). Please find this in the next commit.

Reply:
Changed the name and some functionality with the new optimizations.